pom.xml

```diff
@@ -49,13 +49,8 @@
       </dependency>
       <dependency>
         <groupId>com.hazelcast</groupId>
-        <artifactId>hazelcast-aws</artifactId>
-        <version>3.4</version>  <!-- https://github.com/hazelcast/hazelcast-aws#requirements -->
-      </dependency>
-      <dependency>
-        <groupId>com.hazelcast</groupId>
-        <artifactId>hazelcast-kubernetes</artifactId>
-        <version>2.2.3</version>  <!-- https://github.com/hazelcast/hazelcast-kubernetes#requirements-and-recommendations -->
+        <artifactId>hazelcast</artifactId>
+        <version>5.2.3</version>
```

Comment:
> It seems that vertx-hazelcast is not compatible with hazelcast 5 because ServiceProviderTest.testDiscovery fails: https://github.com/vert-x3/vertx-hazelcast/actions/runs/4927581504/jobs/8804762103?pr=176

Could it be an issue with that test itself?

Reply:
> > Illegal reflective access by com.hazelcast.internal.networking.nio.SelectorOptimizer (file:/home/runner/.m2/repository/com/hazelcast/hazelcast/5.2.3/hazelcast-5.2.3.jar) to field sun.nio.ch.SelectorImpl.selectedKeys
> 
> This will fail in Java 17. The modulepath configuration needs to be fixed: https://docs.hazelcast.com/imdg/4.2/installation/running-in-modular-java

Okapi currently uses **Java 11** not Java 17. When it is time for Okapi to upgrade to Java 17, maybe vertx will have newer version of Hazelcast integrated by then. I did run `mvn install` with Java 17 and it went through fine for now. If there will be issues later, we can deal with that then.